lambdas/build-stac/utils/stac.py

```diff
@@ -47,14 +55,14 @@ def create_stac_item():
 
     if role_arn := os.environ.get("EXTERNAL_ROLE_ARN"):
         creds = role.assume_role(role_arn, "veda-data-pipelines_build-stac")
-        with rasterio.Env(session=AWSSession(
+        gdal_env["session"] = AWSSession(
             aws_access_key_id=creds["AccessKeyId"],
             aws_secret_access_key=creds["SecretAccessKey"],
             aws_session_token=creds["SessionToken"],
-        )):
-            return create_stac_item()
+        )
 
-    return create_stac_item()
+    with rasterio.Env(**gdal_env):
```

Comment:
it was done mostly because the session key is conditional, but i can definitely move these GDAL args down here [8cb00e2](https://github.com/NASA-IMPACT/veda-data-pipelines/pull/183/commits/8cb00e24721483f8f95cb7851331d1ebbd7067fa)

Reply:
Ah, I missed that! Thanks.